src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -81,6 +81,11 @@ public abstract class NodeModel : ModelBase, IRenderPackageSource<NodeModel>, ID
         /// </summary>
         internal HashSet<NodeModel> UpstreamCache = new HashSet<NodeModel>();
 
+        public virtual string NodeType
+        {
+            get { return "ExtensionNode"; }
```

Comment:
Very minor: inconsistent formatting of this function.
All the others do this:

    get  
    {
        return "blah";
    }